ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/execution/versions/electra/ExecutionRequestsDataCodec.java

```diff
@@ -0,0 +1,110 @@
+
+    executionRequestData.forEach(
+        data -> {
+          // First byte of the data is the type of the execution request
+          final byte type = data.get(0);
+          switch (type) {
+            case DepositRequest.REQUEST_TYPE:
+              // https://eips.ethereum.org/EIPS/eip-6110
+              executionRequestsBuilder.deposits(
+                  executionRequestsSchema
+                      .getDepositRequestsSchema()
+                      .sszDeserialize(data.slice(1))
+                      .asList());
+              break;
+            case WithdrawalRequest.REQUEST_TYPE:
+              // https://eips.ethereum.org/EIPS/eip-7002
+              executionRequestsBuilder.withdrawals(
+                  executionRequestsSchema
+                      .getWithdrawalRequestsSchema()
+                      .sszDeserialize(data.slice(1))
+                      .asList());
+              break;
+            case ConsolidationRequest.REQUEST_TYPE:
+              // https://eips.ethereum.org/EIPS/eip-7251
+              executionRequestsBuilder.consolidations(
+                  executionRequestsSchema
+                      .getConsolidationRequestsSchema()
+                      .sszDeserialize(data.slice(1))
+                      .asList());
+              break;
+            default:
+              throw new IllegalArgumentException("Invalid execution request type: " + type);
+          }
+        });
+
+    return executionRequestsBuilder.build();
+  }
+
+  @VisibleForTesting
+  List<Bytes> encode(final ExecutionRequests executionRequests) {
+    final SszList<DepositRequest> depositRequestsSszList =
+        executionRequestsSchema
+            .getDepositRequestsSchema()
+            .createFromElements(executionRequests.getDeposits());
+    final SszList<WithdrawalRequest> withdrawalRequestsSszList =
+        executionRequestsSchema
+            .getWithdrawalRequestsSchema()
+            .createFromElements(executionRequests.getWithdrawals());
+    final SszList<ConsolidationRequest> consolidationRequestsSszList =
+        executionRequestsSchema
+            .getConsolidationRequestsSchema()
+            .createFromElements(executionRequests.getConsolidations());
+
+    return List.of(
+        Bytes.concatenate(
+            Bytes.of(DepositRequest.REQUEST_TYPE), depositRequestsSszList.sszSerialize()),
+        Bytes.concatenate(
+            Bytes.of(WithdrawalRequest.REQUEST_TYPE), withdrawalRequestsSszList.sszSerialize()),
+        Bytes.concatenate(
+            Bytes.of(ConsolidationRequest.REQUEST_TYPE),
+            consolidationRequestsSszList.sszSerialize()));
+  }
+
+  public Bytes32 hash(final ExecutionRequests executionRequests) {
+    final Bytes sortedEncodedRequests =
+        encode(executionRequests).stream()
+            // Encoded requests are sorted by their type (first byte)
+            .sorted(Comparator.comparingInt(b -> b.get(0)))
```

Comment:
do we need to sort it here? haven't we already assembled the list sorted in encode?

Reply:
Yes, but it is a safeguard :) The performance impact is minimal given that it won't really need to re-order anything in the array.